helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -201,7 +201,7 @@ private void createLiveInstance() {
     do {
       retry = false;
       try {
-        _zkclient.createEphemeral(liveInstancePath, liveInstance.getRecord());
+        _zkclient.createEphemeral(liveInstancePath, liveInstance.getRecord(), _sessionId);
```

Comment:
Question: this `createLiveInstance` method is supposed to check if the `_sessionId` is still valid or not. Could you point out where we're doing the check (besides the one happening when we catch `ZkNodeExistsException`)?

Also, is it possible that `_sessionId` is stale? For example, while the ParticipantManager is in the middle of `handleNewSession`, session expired?

Reply:
Good question. 
- The validation is put in `create(path, dataObject, acl, mode, expectedSessionId)` (which is called by `createEphemeral(path, data, session)`):
https://github.com/apache/helix/blob/662dc0f14ed1db49f2c1a41bf55c7cfcecbc7241/helix-core/src/main/java/org/apache/helix/manager/zk/zookeeper/ZkClient.java#L662-L670

   And this is the reason of current PR.

- _sessionId is possibly stale. So we want to fix it in this PR by calling the new API with session id. The new API does final check and create ephemeral node.

Reply:
Great. Could we add a quick comment here:

"_sessionId will be validated in createEphemeral"

Reply:
Done.

Reply:
Done